community/index/src/main/java/org/neo4j/index/internal/gbptree/CrashGenerationCleaner.java

```diff
@@ -74,7 +74,12 @@
     private static long batchSize( long pagesToClean, int threads )
     {
         // Batch size at most maxBatchSize, at least minBatchSize and trying to give each thread 100 batches each
-        return min( MAX_BATCH_SIZE, max( MIN_BATCH_SIZE, pagesToClean / (100 * threads) ) );
```

Comment:
I think a better fix would be to do `(100L * threads)` and avoiding integer multiplication altogether.

Reply:
I'd agree! Good catch.